lib/workers/repository/config-migration/branch/commit-message.ts

```diff
@@ -1,20 +1,23 @@
 import type { RenovateConfig } from '../../../../config/types';
+import { clone } from '../../../../util/clone';
 import * as template from '../../../../util/template';
 import type { CommitMessage } from '../../model/commit-message';
 import { CommitMessageFactory } from '../../model/commit-message-factory';
 
-export class ConfigMigrationCommitMessageFactory {
+export class ConfigMigrationSemanticFactory {
   private readonly config: RenovateConfig;
+  private commitMessage: string | null = null;
+  private prTitle: string | null = null;
 
-  private readonly configFile: string;
-
-  constructor(config: RenovateConfig, configFile: string) {
-    this.config = config;
-    this.configFile = configFile;
+  constructor(config: RenovateConfig, private readonly configFile: string) {
+    this.config = clone(config);
```

Comment:
why do we need to clone now?

Reply:
because the config is being modified between tests runs which caused some to fail. 
therefore i thought it might be a good idea to not modify the original config. 

having said that, the config migration is done last so this might not be a problem.

reverted.

Reply:
This of now has side effects on the config arg, therefore sequential calls to the factory may not work as expected. in other words the class methods are "impure functions" because they modify the config.